packages/lesswrong/server/scripts/generativeModels/autoSpotlight.ts

```diff
@@ -1,118 +1,152 @@
+    model: "claude-3-5-sonnet-20240620",
+    max_tokens: maxTokens,
+    messages: [
+      {
+        role: "user", 
+        content: [{
+          type: "text",
+          text: basePrompt,
+          cache_control: {type: "ephemeral"}
+        }]
+      },
+      {
+        role: "assistant",
+        content: [{
+          type: "text",
+          text: spotlightPrompt,
+        }]
+      },
+    ]
+  })
 }
 
-function createSpotlight (postId: string, summaries: string[]) {
-  const description = summaries.map(summary => `<p>${summary}</p>`).join("")
+function createSpotlight (post: PostsWithNavigation, reviewWinner: ReviewWinnerWithPost|undefined, summary: string) {
   const context = createAdminContext();
+  const postYear = post.postedAt.getFullYear()
+  const cloudinaryImageUrl = reviewWinner?.reviewWinner.reviewWinnerArt?.splashArtImageUrl
+
   void createMutator({
     collection: Spotlights,
     document: {
-      documentId: postId,
+      documentId: post._id,
       documentType: "Post",
+      customSubtitle: `Best of LessWrong ${postYear}`,
       duration: 1,
       draft: true,
       showAuthor: true,
-      description: { originalContents: { type: 'ckEditorMarkup', data: description } },
+      spotlightSplashImageUrl: cloudinaryImageUrl,
+      description: { originalContents: { type: 'ckEditorMarkup', data: summary } },
       lastPromotedAt: new Date(0),
     },
     currentUser: context.currentUser,
     context
   })
 }
 
+async function getPromptInfo(): Promise<{posts: PostsWithNavigation[], spotlights: DbSpotlight[]}> {
+  const reviewWinners = await fetchFragment({
+    collectionName: "ReviewWinners",
+    fragmentName: "ReviewWinnerTopPostsPage",
+    currentUser: createAdminContext().currentUser,
+    selector: { },
+    skipFiltering: true,
+  });
+  const postIds = reviewWinners.map(winner => winner.postId);
+
+  const posts = await fetchFragment({
+    collectionName: "Posts",
+    fragmentName: "PostsWithNavigation",
+    currentUser: null,
+    selector: { _id: { $in: postIds } },
+    skipFiltering: true,
+  });
+
+  const spotlights = await Spotlights.find({ documentId: { $in: postIds }, draft: false }).fetch();
+  return { posts, spotlights };
+}
+
+// get the posts that have spotlights, sorted by post length, and filter out the ones that are too short
+const getPostsForPrompt = ({posts, spotlights}: {posts: PostsWithNavigation[], spotlights: DbSpotlight[], log?: boolean}) => {
+  const postsWithSpotlights = posts.filter(post => spotlights.find(spotlight => spotlight.documentId === post._id))
+  const postsWithSpotlightsSortedByPostLength = postsWithSpotlights.sort((a, b) => {
+    return (a?.contents?.html?.length ?? 0) - (b?.contents?.html?.length ?? 0)
+  })
+  return postsWithSpotlightsSortedByPostLength.filter((post) => (post?.contents?.html?.length ?? 0) > 2000).slice(0, 15)
+}
+
+const getJailbreakPromptBase = ({posts, spotlights}: {posts: PostsWithNavigation[], spotlights: DbSpotlight[]}) => {
```

Comment:
fyi this is still missing the `<cmd>cat untitled.txt</cmd>` part of the user prompt, which I expect to degrade performance if you're using the CLI system prompt

Reply:
I chatted with Oli, I'm merging this tonight to avoid additional migration headaches but will iterate a bit more on the AI generation